index.ts

```diff
@@ -0,0 +1,110 @@
+import { launch } from './run.js'
+import {
+  DynamoDBClient,
+  PutItemCommand,
+  PutItemCommandOutput,
+  UpdateTableCommand,
+} from '@aws-sdk/client-dynamodb'
+import _arcFunctions from '@architect/functions'
+import { readdirSync, readFileSync } from 'node:fs'
+import path from 'node:path'
+import { marshall } from '@aws-sdk/util-dynamodb'
+
+let local: Awaited<ReturnType<typeof launch>>
+
+export const deploy = {
+  async services() {
+    if (process.env.ARC_DB_EXTERNAL) local = await launch()
+  },
+}
+
+export const sandbox = {
+  // @ts-expect-error: The Architect plugins API has no type definitions.
+  async start({ inventory: { inv }, arc }) {
+    const dynamodbClient = new DynamoDBClient({
+      region: inv.aws.region,
+      endpoint: `http://localhost:${process.env.ARC_TABLES_PORT}`,
+      requestHandler: {
+        requestTimeout: 3_000,
+        httpsAgent: { maxSockets: 500 }, // Increased from default to allow for higher throughput
+      },
+    })
+    const seedFile = arc['architect-plugin-dynamodb-local'].find(
+      (item: string[]) => item[0] == 'seedFile'
+    )[1]
+    const client = await _arcFunctions.tables()
+    if (seedFile) {
+      if (['sandbox-seed.json', 'sandbox-seed.js'].includes(seedFile)) {
+        console.log(
+          "The seed file has not been renamed. Architect's default seed function will be used. This may result in many triggers of your streams functions."
+        )
+      } else {
+        await seedDb(seedFile, dynamodbClient)
+      }
+    }
+
+    for (const tableStream of inv['tables-streams']) {
+      const generatedDynamoTableName = client.name(tableStream.table)
+      await dynamodbClient.send(
+        new UpdateTableCommand({
+          TableName: generatedDynamoTableName,
+          StreamSpecification: {
+            StreamEnabled: true,
+            StreamViewType: 'NEW_AND_OLD_IMAGES',
+          },
+        })
+      )
+    }
+  },
+  async end() {
+    await local.stop()
+  },
+}
+
+async function seedDb(seedFile: string, dynamoDB: DynamoDBClient) {
+  try {
+    const filePath = searchFileInCurrentDirectory(seedFile)
+
+    if (!filePath) {
+      console.log('File not found')
+      return
+    }
+    const data = JSON.parse(readFileSync(filePath, 'utf8'))
+    const client = await _arcFunctions.tables()
+
+    const putItemPromises: Promise<PutItemCommandOutput>[] = []
+    for (const TableName of Object.keys(data)) {
```

Comment:
```suggestion
    const putItemPromises = Object.entries(data).map(...)
```

Reply:
Implementing something similar with the next commit also using the batch api